Migrate `@media screen(…)`

RobinMalfait opened the pull request:
This PR adds a codemod that migrates the `@media screen(…)` to the properly expanded `@media (…)` syntax.

```css
@media screen(md) {
  .foo {
    color: red;
  }
}
```

Will be converted to:
```css
@media (width >= 48rem) {
  .foo {
    color: red;
  }
}
```

If you happen to have custom screens (even complex ones), the screen will be converted to a custom media query.

```css
@media screen(foo) {
  .foo {
    color: red;
  }
}
```
With a custom `tailwind.config.js` file with a config like this:
```js
module.exports = {
  // …
  theme: {
    screens: {
      foo: { min: '100px', max: '123px' },
    },
  }
}
```

Then the codemod will convert it to:
```css
@media (123px >= width >= 100px) {
  .foo {
    color: red;
  }
}
```

RobinMalfait committed:
migrate `screen(…)`

RobinMalfait committed:
update changelog

thecrypticace reviewed (approved):
💪

adamwathan reviewed (dismissed):
Nice! I think this should use the theme function instead of inlining the raw value though.

RobinMalfait committed:
if we resolve to a string, use the `theme(…)` function

In other cases, or using legacy JS object syntax, we can't use the
`theme(…)` function, therefore we fully expand the media query in your
CSS.

RobinMalfait committed:
update tests

RobinMalfait committed:
alternative solution: `@custom-media`

RobinMalfait committed:
Revert "alternative solution: `@custom-media`"

This reverts commit a1113f837b738a360da5d31b145771ad329be72a.

This is way too experimental and no browser implements this feature yet.
See: https://caniuse.com/?search=custom-media

We can use lightningcss that transpiles is properly, but in environments
such as the CDN we don't have access to lightningcss.

RobinMalfait commented:
Made some updates:

1. If we can use the `theme(--breakpoint-{name})` function, then we will use that:
   https://github.com/tailwindlabs/tailwindcss/blob/28b2c1a1e0afbe3e37aa3c3d17d35562e4564512/packages/%40tailwindcss-upgrade/src/codemods/migrate-media-screen.test.ts#L26-L42
3. If we cannot, then we will expand the full media query.
   https://github.com/tailwindlabs/tailwindcss/blob/28b2c1a1e0afbe3e37aa3c3d17d35562e4564512/packages/%40tailwindcss-upgrade/src/codemods/migrate-media-screen.test.ts#L152-L177

Also tried with an alternative using `@custom-media`, but since we can't use that I reverted it. You can see the changes here: https://github.com/tailwindlabs/tailwindcss/pull/14603/commits/a1113f837b738a360da5d31b145771ad329be72a
